src/Infrastructure/DependencyInjection.cs

```diff
@@ -12,18 +20,44 @@ public static class DependencyInjection
 {
     public static IServiceCollection AddInfrastructureServices(this IServiceCollection services, IConfiguration configuration)
     {
-        //Refit Playfab Settings
+        SetupRefitHttpClients(services, configuration);
+
+        services.AddSingleton<IEventStreamDataProcessor<JsonElement>, PlayfabEventStreamDataProcessor>();
+        
+        //Providers are usually ThirdParty services
+        services.AddSingleton<IMetricProvider, PlayfabMetricProvider>();
+        
+        //Register Named QuestProvider using it's Handler as KeyedName
+        services.AddKeyedSingleton<IQuestProvider, YggQuestProvider>(typeof(YggQuestProviderHandler));
+        
+        services.AddScoped<ISecureDataService, SecureDataService>();
+        services.AddLogging();
+        
+        return services;
+    }
+
+    private static void SetupRefitHttpClients(IServiceCollection services, IConfiguration configuration)
+    {
         services.AddRefitClient<IPlayfabAPI>()
             .ConfigureHttpClient(c =>
             {
                 c.BaseAddress = new Uri($"https://{configuration.GetValue<string>("AppSettings:Playfab:PlayfabTitleId")}.playfabapi.com");
                 c.DefaultRequestHeaders.Add("X-SecretKey", configuration.GetValue<string>("AppSettings:Playfab:PlayfabSecretKey"));
             });
         
-        services.AddScoped<IMetricProvider, PlayfabMetricProvider>();
-        services.AddScoped<ISecureDataService, SecureDataService>();
-        
-        return services;
+        services.AddRefitClient<IYggAPI>()
+            .ConfigureHttpClient(c =>
+            {
+                var baseUrl = configuration.GetValue<string>("AppSettings:QuestProvider:Ygg:Endpoint");
+
+                if (baseUrl.IsNullOrEmpty())
+                {
+                    throw new ArgumentException("Error setting up QuestProvider: YGG, Invalid/Null Base Endpoint");
+                }
+
+                c.BaseAddress = new Uri(baseUrl!);
+                c.DefaultRequestHeaders.Add("API_KEY", configuration.GetValue<string>("AppSettings:QuestProvider:Ygg:ApiKey"));
+            });

```

Comment:
Pretty cool way of declaring API's ! Really liked it !

Reply:
Refit is awesome!